### PR TITLE
Simplify finding metrics in Summarize

### LINKED ISSUE #46843: Common Metrics search doesn't work when you add a space ` `
### Describe the bug

Common Metrics search doesn't work when you add a space ` `

### To Reproduce

Create a bunch of metrics having a space in their name and try to search for them:

https://github.com/user-attachments/assets/208c7bc1-1770-4aaa-97b9-2e0ead52d999

### Expected behavior

I am able to search for names which have a space in their name

### Logs

None

### Information about your Metabase installation

```JSON
master and version 50
```

### Severity

annoying

### Additional context

_No response_

### LINKED ISSUE #46847: [Epic] Simplify finding metrics in Summarize
[product doc](https://www.notion.so/metabase/Simplify-finding-metrics-in-Summarize-7f70c572fa5640a2a76caa4ee8625ca4)
[testing plan](https://www.notion.so/metabase/Testing-plan-Simplifying-metrics-in-Summarize-692ca2a9c0154e948ce3a36cb2c62c95)

- Always show the input field on top of the Summarize block.
- On hover in the list of Common Metrics show (i) icon and on hover on this icon a tooltip with a description for this metric. When clicking on the icon open the metric.
- Make operators searchable in aggregation picker
- Sort the available metrics

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/46847
Closes https://github.com/metabase/metabase/issues/46843

### Description

- Use global search (searchbar on top) for the aggregation picker
- Allow operators and metrics to be search (and fix https://github.com/metabase/metabase/issues/46843)
- Add info icon for metrics in the aggregation picker
- Sort the available metrics

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Create a metric (ie. New -> Metric -> Orders -> Aggregate -> Count -> Save -> name: `Metric Foo`, description: `This is a description`)
2. New -> Question -> Orders
3. Create a new aggregation
4. Verify that the search bar is visible and on top
5. In the search bar, type `Count`. Verify that the filtering works correctly and only shows relevant matches
6. In the search bar, type (part of) type `Foo`. Verify that your metric is visible.
7. Hover the metric, an info icon should appear.
8. Hover the info icon, your description should show in a card.
9. Create another metric and call it `AAA`
10. New -> Question -> Orders
11. Create a new aggregation 
12. Verify that the metrics are sorted

### Demo

https://github.com/user-attachments/assets/f558b293-6ba2-4d3b-90d6-559fca496c0e

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx [hunk 1]
```diff
@@ -365,6 +377,31 @@ function renderItemName(item: ListItem) {
...
+  if (item.type !== "metric") {
+    return null;
+  }
+
+  if (!item.description) {
+    return null;
+  }
+
+  return (
+    <PopoverWrapper>
```
Comment (ranquild) [inline frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx:394]: `Flex` from `metabase/ui` should work. We generally don't add new styled components now. Either mantine or CSS

File: src/metabase/lib/metric.cljc [hunk 2]
```diff
@@ -127,4 +127,4 @@
                 (comp (filter (fn [metric-card]
                                 (= 1 (lib.query/stage-count (lib.query/query query (:dataset-query metric-card))))))
                       (map maybe-add-aggregation-pos))
-                metrics)))))))
+                (sort-by :display-name metrics))))))))
```
Comment (ranquild) [inline src/metabase/lib/metric.cljc:130]: Can you add a simple clojure test for this?

**Review by ranquild — CHANGES_REQUESTED** [review]
I have a metric with a description but it isn't shown in the popover:
Screenshot 2024-08-15 at 09 57 50
Screenshot 2024-08-15 at 09 57 45

Also - do you handle markdown in description? We do something special in this case, at least in collections

**Comment by romeovs** [comment]
@ranquild 

> I have a metric with a description but it isn't shown in the popover

Strange I cannot reproduce that, all my metrics with descriptions show the icon... Can you tell me what the value of `item.description` is on [this line](https://github.com/metabase/metabase/pull/46887/files#diff-f398a211ca1bf3802d002be844d765031e6bfef79264d34fc4f91763121f36c8R388) for those metrics?

**Comment by romeovs** [comment]
@ranquild I've added a test but I'm unsure why it's breaking all the builds!

File: frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx [hunk 3]
```diff
@@ -365,6 +377,38 @@ function renderItemName(item: ListItem) {
...
+    return null;
+  }
+
+  return (
+    <Flex pr="sm" align="center">
+      <Popover
+        position="right"
+        content={
+          <Box p="md">
+            <Markdown>{item.description}</Markdown>
```
Comment (ranquild) [inline frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx:399]: We should probably use `disallowHeading` and `unstyleLinks` as we do in similar places https://github.com/metabase/metabase/blob/dfab26710da2b14bf1a1b43c4e941f396d3f9ec6/frontend/src/metabase/visualizations/components/ScalarValue/ScalarValue.jsx#L105. Could you confirm this with the product? Could you add tests for these cases?

**Review by ranquild — COMMENTED** [review]
`item.description` is shown now. I have several questions about markdown support for which I left in a comment. Let's figure it out before merging this.

**Comment by ranquild** [comment]
@metabase/core-backend-querying please review the mbql lib changes

File: frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx (hunk 3 above)
Comment (romeovs) [reply frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx]: @mngr Can you confirm we want to render this as Markdown without headings and with unstyled links? We render de description as such elsewhere so I think it makes sense.

Comment (mngr) [reply frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx]: > @mngr Can you confirm we want to render this as Markdown without headings and with unstyled links? We render de description as such elsewhere so I think it makes sense.

Yes, let's do that

**Review by lbrdnk — APPROVED** [review]
BE part looks OK.

File: test/metabase/lib/metric_test.cljc [hunk 4]
```diff
@@ -211,6 +231,13 @@
                     :aggregation-position 0}]
                   (map #(lib/display-info query-with-metric %)
                        metrics)))))))
+
+  (testing "Should return the available metrics as sorted"
+    (let [query   (lib/query metadata-provider-with-multiple-metrics (meta/table-metadata :venues))
+          metrics (lib.metric/available-metrics query)]
+      (is (=? (:id (first metrics)) second-metric-id))
+      (is (=? (:id (second metrics)) metric-id))))
```
Comment (lbrdnk) [inline test/metabase/lib/metric_test.cljc:239]: nit: I believe `(=? [{:id second-metric-id} {:id metric-id}] metrics)` could be used here.

**Review by ranquild — APPROVED** [review]

**Pull request merged**